### PR TITLE
fix python requirement to >=3.8

### LINKED ISSUE #54: 1.31 not compatible with py3.7
### Solution to issue cannot be found in the documentation.

- [x] I checked the documentation.

### Issue

```
mamba create -n repro azure-core=1.31 python=3.7
```
and then
```
python -c 'import azure.core.credentials'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/aopfer/mambaforge/envs/repro/lib/python3.7/site-packages/azure/core/__init__.py", line 31, in <module>
    from ._pipeline_client import PipelineClient
  File "/home/aopfer/mambaforge/envs/repro/lib/python3.7/site-packages/azure/core/_pipeline_client.py", line 31, in <module>
    from .pipeline import Pipeline
  File "/home/aopfer/mambaforge/envs/repro/lib/python3.7/site-packages/azure/core/pipeline/__init__.py", line 180, in <module>
    from ._base import Pipeline  # pylint: disable=wrong-import-position
  File "/home/aopfer/mambaforge/envs/repro/lib/python3.7/site-packages/azure/core/pipeline/_base.py", line 34, in <module>
    from azure.core.pipeline.policies import HTTPPolicy, SansIOHTTPPolicy
  File "/home/aopfer/mambaforge/envs/repro/lib/python3.7/site-packages/azure/core/pipeline/policies/__init__.py", line 28, in <module>
    from ._authentication import (
  File "/home/aopfer/mambaforge/envs/repro/lib/python3.7/site-packages/azure/core/pipeline/policies/_authentication.py", line 8, in <module>
    from azure.core.credentials import TokenCredential, SupportsTokenInfo, TokenRequestOptions, TokenProvider
  File "/home/aopfer/mambaforge/envs/repro/lib/python3.7/site-packages/azure/core/credentials.py", line 6, in <module>
    from typing import Any, NamedTuple, Optional, TypedDict, Union, ContextManager
ImportError: cannot import name 'TypedDict' from 'typing' (/home/aopfer/mambaforge/envs/repro/lib/python3.7/typing.py)
```
Downgrade to 1.30.2 solves the problem

### Installed packages
```
# packages in environment at /home/aopfer/mambaforge/envs/repro:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
azure-core                1.31.0             pyhd8ed1ab_0    conda-forge
brotli-python             1.0.9            py37hd23a5d3_7    conda-forge
ca-certificates           2024.8.30            hbcca054_0    conda-forge
certifi                   2024.8.30          pyhd8ed1ab_0    conda-forge
charset-normalizer        3.4.0              pyhd8ed1ab_0    conda-forge
idna                      3.10               pyhd8ed1ab_0    conda-forge
ld_impl_linux-64          2.43                 h712a8e2_2    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc                    14.2.0               h77fa898_1    conda-forge
libgcc-ng                 14.2.0               h69a702a_1    conda-forge
libgomp                   14.2.0               h77fa898_1    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsqlite                 3.47.0               hadc24fc_1    conda-forge
libstdcxx                 14.2.0               hc0a3c3a_1    conda-forge
libstdcxx-ng              14.2.0               h4852527_1    conda-forge
libzlib                   1.3.1                hb9d3cd8_2    conda-forge
ncurses                   6.5                  he02047a_1    conda-forge
openssl                   3.3.2                hb9d3cd8_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1            py37h89c1867_5    conda-forge
python                    3.7.12          hf930737_100_cpython    conda-forge
python_abi                3.7                     4_cp37m    conda-forge
readline                  8.2                  h8228510_1    conda-forge
requests                  2.32.2             pyhd8ed1ab_0    conda-forge
setuptools                69.0.3             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.47.0               h9eae976_1    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
typing-extensions         4.7.1                hd8ed1ab_0    conda-forge
typing_extensions         4.7.1              pyha770c72_0    conda-forge
urllib3                   2.2.1              pyhd8ed1ab_0    conda-forge
wheel                     0.42.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge

```
### Environment info
```
     active environment : repro
    active env location : /home/aopfer/mambaforge/envs/repro
            shell level : 1
       user config file : /home/aopfer/.condarc
 populated config files : /etc/conda/condarc
                          /home/aopfer/.condarc
          conda version : 24.5.0
    conda-build version : not installed
         python version : 3.10.14.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=zen3
                          __conda=24.5.0=0
                          __cuda=12.2=0
                          __glibc=2.36=0
                          __linux=6.9.10=0
                          __unix=0=0
       base environment : /home/aopfer/mambaforge  (writable)
      conda av data dir : /home/aopfer/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : http://lxc-nginx-internal.lan/conda/conda-forge/linux-64
                          http://lxc-nginx-internal.lan/conda/conda-forge/noarch
          package cache : /home/aopfer/mambaforge/pkgs
                          /home/aopfer/.conda/pkgs
       envs directories : /home/aopfer/mambaforge/envs
                          /home/aopfer/.conda/envs
               platform : linux-64
             user-agent : conda/24.5.0 requests/2.31.0 CPython/3.10.14 Linux/6.9.10+bpo-amd64 debian/12.7 glibc/2.36 solver/libmamba conda-libmamba-solver/24.1.0 libmambapy/1.5.8
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by AaronOpfer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #54

### CODE REVIEW

**Comment by AaronOpfer** [comment]
@conda-forge-admin, please rerender

**Comment by AaronOpfer** [comment]
@ocefpaf , did c-f decide on a technique for dealing with the issue of old Python versions picking up new recipes, that I should add here?

**Comment by ocefpaf** [comment]
> @ocefpaf , did c-f decide on a technique for dealing with the issue of old Python versions picking up new recipes, that I should add here?

Actually, we just voted and the core team is wroking on docs, migrations, and updates. See https://github.com/conda-forge/conda-forge.github.io/pull/2355

**Comment by AaronOpfer** [comment]
@beckermr could you review for me? Maintainer appears AFK. I plan on getting a repodata patch going soon after.

**Comment by beckermr** [comment]
@conda-forge-admin relint

**Comment by beckermr** [comment]
Please address the hints here.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,29 @@
 {% set name = "azure-core" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml:1]:
```suggestion
{% set python_min = "3.9" %}
{% set name = "azure-core" %}
```

**Pull request merged**